gc/base/SweepHeapSectioning.cpp

```diff
@@ -339,3 +346,90 @@ MM_SweepHeapSectioning::getBackingStoreSize()
 {
 	return _baseArray->_used * sizeof(MM_ParallelSweepChunk);
 }
+
+uintptr_t
+MM_SweepHeapSectioning::reassignChunks(MM_EnvironmentBase *env)
+{
+	MM_ParallelSweepChunk *chunk; /* Sweep table chunk (global) */
+	MM_ParallelSweepChunk *previousChunk;
+	uintptr_t totalChunkCount;  /* Total chunks in system */
+
+	MM_SweepHeapSectioningIterator sectioningIterator(this);
+
+	totalChunkCount = 0;
+	previousChunk = NULL;
+
+	MM_HeapRegionManager *regionManager = _extensions->getHeap()->getHeapRegionManager();
+	GC_HeapRegionIterator regionIterator(regionManager);
+	MM_HeapRegionDescriptor *region = NULL;
+
+	while (NULL != (region = regionIterator.nextRegion())) {
+		if (isReadyToSweep(env, region)) {
+			/* TODO:  this must be rethought for Tarok since it treats all regions identically but some might require different sweep logic */
+			uintptr_t *heapChunkBase = (uintptr_t *)region->getLowAddress();  /* Heap chunk base pointer */
+			uintptr_t *regionHighAddress = (uintptr_t *)region->getHighAddress();
+
+			while (heapChunkBase < regionHighAddress) {
+				void *poolHighAddr;
+				uintptr_t *heapChunkTop;
+				MM_MemoryPool *pool;
+
+				chunk = sectioningIterator.nextChunk();
+				Assert_MM_true(chunk != NULL);  /* Should never return NULL */
+				totalChunkCount += 1;
+
+				/* Clear all data in the chunk (including sweep implementation specific information) */
+				chunk->clear();
+
+				if(((uintptr_t)regionHighAddress - (uintptr_t)heapChunkBase) < _extensions->parSweepChunkSize) {
+					/* corner case - we will wrap our address range */
+					heapChunkTop = regionHighAddress;
+				} else {
+					/* normal case - just increment by the chunk size */
+					heapChunkTop = (uintptr_t *)((uintptr_t)heapChunkBase + _extensions->parSweepChunkSize);
+				}
+
+				/* Find out if the range of memory we are considering spans 2 different pools.  If it does,
+				 * the current chunk can only be attributed to one, so we limit the upper range of the chunk
+				 * to the first pool and will continue the assignment at the upper address range.
+				 */
+				pool = region->getSubSpace()->getMemoryPool(env, heapChunkBase, heapChunkTop, poolHighAddr);
+				if (NULL == poolHighAddr) {
+					heapChunkTop = (heapChunkTop > regionHighAddress ? regionHighAddress : heapChunkTop);
+				} else {
+					/* Yes ..so adjust chunk boundaries */
+					assume0(poolHighAddr > heapChunkBase && poolHighAddr < heapChunkTop);
```

Comment:
Probably does not make sense to keep `assume0()` calls. It should be converted to real assertion or removed. Also there is another one below